plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMetadataFileOperations.java

```diff
@@ -75,90 +86,238 @@ protected DistributedQueryRunner createQueryRunner()
+        assertUpdate("CREATE TABLE test_join_t1 AS SELECT 2 as age, 'id1' AS id", 1);
+        assertUpdate("CREATE TABLE test_join_t2 AS SELECT 'name1' as name, 'id1' AS id", 1);
 
-        public OperationCounts(Map<OperationContext, Integer> allCounts)
+        assertFileSystemAccesses("SELECT name, age FROM test_join_t1 JOIN test_join_t2 ON test_join_t2.id = test_join_t1.id",
+                ImmutableMultiset.builder()
+                        .addCopies(new FileOperation(MANIFEST, INPUT_FILE_GET_LENGTH), 12)
+                        .addCopies(new FileOperation(MANIFEST, INPUT_FILE_NEW_STREAM), 12)
+                        .addCopies(new FileOperation(METADATA_JSON, INPUT_FILE_NEW_STREAM), 2)
+                        .addCopies(new FileOperation(SNAPSHOT, INPUT_FILE_GET_LENGTH), 2)
+                        .addCopies(new FileOperation(SNAPSHOT, INPUT_FILE_NEW_STREAM), 2)
+                        .build());
+    }
+
+    @Test
+    public void testJoinWithPartitionedTable()
+            throws Exception
+    {
+        assertUpdate("CREATE TABLE test_join_partitioned_t1 (a BIGINT, b TIMESTAMP(6) with time zone) WITH (partitioning = ARRAY['a', 'day(b)'])");
+        assertUpdate("CREATE TABLE test_join_partitioned_t2 (foo BIGINT)");
+        assertUpdate("INSERT INTO test_join_partitioned_t2 VALUES(123)", 1);
+        assertUpdate("INSERT INTO test_join_partitioned_t1 VALUES(123, current_date)", 1);
+
+        assertFileSystemAccesses("SELECT count(*) FROM test_join_partitioned_t1 join test_join_partitioned_t2 on test_join_partitioned_t1.a = test_join_partitioned_t2.foo",
+                ImmutableMultiset.builder()
+                        .addCopies(new FileOperation(MANIFEST, INPUT_FILE_GET_LENGTH), 12)
+                        .addCopies(new FileOperation(MANIFEST, INPUT_FILE_NEW_STREAM), 12)
+                        .addCopies(new FileOperation(METADATA_JSON, INPUT_FILE_NEW_STREAM), 2)
+                        .addCopies(new FileOperation(SNAPSHOT, INPUT_FILE_GET_LENGTH), 2)
+                        .addCopies(new FileOperation(SNAPSHOT, INPUT_FILE_NEW_STREAM), 2)
+                        .build());
+    }
+
+    @Test
+    public void testExplainSelect()
+            throws Exception
+    {
+        assertUpdate("CREATE TABLE test_explain AS SELECT 2 as age", 1);
+
+        assertFileSystemAccesses("EXPLAIN SELECT * FROM test_explain",
+                ImmutableMultiset.builder()
+                        .addCopies(new FileOperation(MANIFEST, INPUT_FILE_GET_LENGTH), 2)
+                        .addCopies(new FileOperation(MANIFEST, INPUT_FILE_NEW_STREAM), 2)
+                        .addCopies(new FileOperation(METADATA_JSON, INPUT_FILE_NEW_STREAM), 1)
+                        .addCopies(new FileOperation(SNAPSHOT, INPUT_FILE_GET_LENGTH), 1)
+                        .addCopies(new FileOperation(SNAPSHOT, INPUT_FILE_NEW_STREAM), 1)
+                        .build());
+    }
+
+    @Test
+    public void testShowStatsForTable()
```

Comment:
add SHOW STATS FOR partitioned table

Reply:
I added one but its result is strange

Reply:
what's strange about them? is it related to the fact the table is empty?
(https://github.com/trinodb/trino/pull/8679#discussion_r679735696)